core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1817,6 +1842,9 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
             log.info("  <-pending");
 
         if (bestChain) {
+            boolean wasDead = dead.remove(txHash) != null;
```

Comment:
yes, receive() is called from several methods including reorganize(), so this is part of the reorganize logic.

In a previous version of this PR, the reorganize "undo" logic was:
1) txA is in conflict with txB -> both are in the pending pool, confidence marked for both of them as "in conflict".
2) txA gets into blockX that is part of the best chain -> txA goes to building, txB goes to dead.
3) blockY that competes with blockX is now the new best block. bloc Y does not includes txA nor txB -> txA and txB go back to the pending pool, confidence marked for both of them as "in conflict".
4) If blockY would have included txB or a block on top of blockY includes txB -> txA goes to dead, txB goes to building.

In that code, all the resurrecting code was part of the "undo" logic in reorganize()

Since you told me dead txs are not normally resurrected, I replaced step "3" with this logic:
3) blockY that competes with blockX is now the new best block. blockY does not includes txA nor txB -> txA goes back to the pending pool, txB remains dead.

Now, step "4" logic remains the same, but the "roll forward" logic (ie receive() method) should support the case where we are receiving in a best chain block, a tx that was dead, so that is why this dead handling code here.